### PR TITLE
Fix flaky TestingKafka tests

### PR DESCRIPTION
**Pull request by elonazoulay**

### CODE REVIEW

**Comment by elonazoulay** [comment]
Fixes #6412

File: plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestKafkaWithConfluentSchemaRegistryMinimalFunctionality.java [hunk 1]
```diff
@@ -73,6 +73,11 @@
             .name("col_3").type().optional().doubleType()
             .endRecord();
 
+    private TestingKafka createWithSchemaRegistry()
```
Comment (Praveen2112) [inline plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestKafkaWithConfluentSchemaRegistryMinimalFunctionality.java:76]: nit : Can we capture this change in a separate commit.. Not mandatory

**Review by Praveen2112 — COMMENTED** [review]
How is that it fixes #6412. If it is due to Classloader issue then it should fail for each time right ? https://github.com/confluentinc/schema-registry/issues/1176 tracks this issue.

File: plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestKafkaWithConfluentSchemaRegistryMinimalFunctionality.java (hunk 1 above)
Comment (elonazoulay) [reply plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestKafkaWithConfluentSchemaRegistryMinimalFunctionality.java:76]: Will look into why this works and update.

**Comment by findepi** [comment]
> Fixes #6412

This particular is closed, but there are two others related https://github.com/trinodb/trino/issues?q=is%3Aissue+is%3Aopen+TestKafkaWithConfluentSchemaRegistryMinimalFunctionality

File: testing/trino-testing-kafka/src/main/java/io/trino/testing/kafka/TestingKafka.java [hunk 2]
```diff
@@ -180,19 +189,21 @@ public void createTopicWithConfig(int partitions, int replication, String topic,
...
-            producer.flush();
-            return future.get();
-        }
-        catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            throw new RuntimeException(e);
-        }
-        catch (ExecutionException e) {
-            throw new RuntimeException(e);
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
```
Comment (findepi) [inline testing/trino-testing-kafka/src/main/java/io/trino/testing/kafka/TestingKafka.java:192]: during test execution from maven there is probably just one classloader, so current TCCL = classLoader

File: plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/KafkaWithConfluentSchemaRegistryQueryRunner.java [hunk 3]
```diff
@@ -15,6 +15,7 @@
```
Comment (kokosing) [inline plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/KafkaWithConfluentSchemaRegistryQueryRunner.java:15]: Can you please explain in commit message what was flaky and why?

File: plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestKafkaWithConfluentSchemaRegistryMinimalFunctionality.java [hunk 4]
```diff
@@ -145,11 +147,10 @@ public void testTopicWithRecordNameStrategy()
     }
 
     @Test
-    @Flaky(issue = "https://github.com/trinodb/trino/issues/6412", match = "Error registering Avro schema: .*")
     public void testTopicWithTopicRecordNameStrategy()
             throws Exception
     {
-        try (TestingKafka testingKafka = TestingKafka.createWithSchemaRegistry();
+        try (TestingKafka testingKafka = createWithSchemaRegistry();
```
Comment (kokosing) [inline plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestKafkaWithConfluentSchemaRegistryMinimalFunctionality.java:153]: If the flaky issue is fixed then we no longer need to setup testing kafka and query runner per each test. Can you please remove it?

**Review by kokosing — COMMENTED** [review]
Please stress the change by running kafka tests suite several times in `.github/workflows/ci.yml` while other tests are removed. It is flaky, so we need bigger gun.

File: plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/KafkaWithConfluentSchemaRegistryQueryRunner.java (hunk 3 above)
Comment (elonazoulay) [reply plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/KafkaWithConfluentSchemaRegistryQueryRunner.java:15]: Will do - and I'll make sure to find out why:) It seems like a suppressed or wrapped error from KafkaAvroSerializer but will verify.

**Comment by kokosing** [comment]
I think we can close it now as underlying issue is now fixed.

**Pull request closed without merging**